certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/certs.kt

```diff
@@ -35,18 +40,29 @@ internal fun String.parsePemCertificate(fileName: String? = null): Certificate {
   return CertificateAdapters.certificate.fromDer(data)
 }
 
-internal fun File.parsePemCertificate(): Certificate {
-  try {
-    val pemText = readText()
+@Suppress("BlockingMethodInNonBlockingContext")
+@OptIn(ExperimentalFilesystem::class)
+internal suspend fun Path.parsePemCertificate(filesystem: Filesystem = Filesystem.SYSTEM): Certificate {
+  return withContext(Dispatchers.IO) {
+    try {
+      val pemText = filesystem.source(this@parsePemCertificate).buffer().use { it.readUtf8() }
 
-    return pemText.parsePemCertificate(name)
-  } catch (fnfe: FileNotFoundException) {
-    throw UsageException("No such file: $this", fnfe)
+      pemText.parsePemCertificate(name)
+    } catch (fnfe: FileNotFoundException) {
+      throw UsageException("No such file: $this", fnfe)
+    }
   }
 }
 
-internal fun X509Certificate.writePem(
-  output: File
+@Suppress("BlockingMethodInNonBlockingContext")
+@OptIn(ExperimentalFilesystem::class)
+internal suspend fun X509Certificate.writePem(
+  output: Path,
+  filesystem: Filesystem = Filesystem.SYSTEM
 ) {
-  output.writeText(certificatePem())
+  withContext(Dispatchers.IO) {
+    filesystem.sink(output).buffer().use {
```

Comment:
I think we probably want `filesystem.writeUtf8(output, certificatePem())`

Reply:
extension methods would be great here.

Reply:
on the flip side.

```
fun Path.writeUtf8(fs: Filesystem = Filesystem.SYSTEM) {}
```

Would be problematic but nice for the defaults.